core/src/task.rs

```diff
@@ -47,6 +68,14 @@ impl AuthenticationToken {
     pub fn as_bytes(&self) -> &[u8] {
         &self.0
     }
+
+    /// Generate a random authentication token.
+    pub fn random() -> Self {
```

Comment:
Instead, `impl Distribution<AuthenticationToken> for Standard`. There are a few examples in the messages crate, among other places in the codebase -- the function body will stay much the same except that you'll end up calling `rng.gen()` instead of `random()` to generate underlying random values.